app/forms/FastPassForm.scala

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import _root_.forms.Mappings._
+import play.api.data.{ Form, FormError }
+import play.api.data.Forms._
+import play.api.data.format.Formatter
+import play.api.i18n.Messages
+
+object FastPassForm {
+
+  val EmptyFastPassDetails: Data = Data("", None, None, None, None)
+
+  val CivilServant = "CivilServant"
+  val CivilServantViaFastTrack = "CivilServantViaFastTrack"
+  val DiversityInternship = "DiversityInternship"
+
+  val EDIP = "EDIP"
+  val SDIPPreviousYear = "SDIPPreviousYear"
+  val SDIPCurrentYear = "SDIPCurrentYear"
+
+  val FastPassTypes = Seq(
+    CivilServant -> Messages("fastPassType.CivilServant"),
+    CivilServantViaFastTrack -> Messages("fastPassType.CivilServantViaFastTrack"),
+    DiversityInternship -> Messages("fastPassType.DiversityInternship")
+  )
+
+  val InternshipTypes = Seq(
+    EDIP -> Messages("internshipType.EDIP"),
+    SDIPPreviousYear -> Messages("internshipType.SDIPPreviousYear"),
+    SDIPCurrentYear -> Messages("internshipType.SDIPCurrentYear")
+  )
+
+  val fastPassTypeRequiredMsg = Messages("error.fastPassType.required")
+  val internshipTypeRequiredMsg = Messages("error.internshipTypes.required")
+  val fastPassReceivedRequiredMsg = Messages("error.fastPassReceived.required")
+  val certificateNumberRequiredMsg = Messages("error.certificateNumber.required")
+
+  val formQualifier = "fastPassDetails"
+  val applicable = "applicable"
+  val fastPassType = "fastPassType"
+  val internshipTypes = "internshipTypes"
+  val fastPassReceived = "fastPassReceived"
+  val certificateNumber = "certificateNumber"
+
+  case class Data(applicable: String,
+                  fastPassType: Option[String] = None,
+                  internshipTypes: Option[Seq[String]] = None,
+                  fastPassReceived: Option[Boolean] = None,
+                  certificateNumber: Option[String] = None)
+
+  def form = {
+    Form(mapping(
+      applicable -> nonemptyBooleanText("error.applicable.required"),
+      fastPassType -> of(fastPassTypeFormatter),
+      internshipTypes -> of(internshipTypesFormatter),
+      fastPassReceived -> of(fastPassReceivedFormatter),
+      certificateNumber -> of(fastPassCertificateFormatter)
```

Comment:
Since it works, we've agreed to let this through and perhaps have a crack at alternative methods at a later date